stencils/Navigation.stencil

```diff
@@ -98,6 +98,58 @@ struct Navigation<NavigationContent: View>: ViewModifier {
     }
 }
 
+struct ItemNavigation<ID : Equatable, NavigationContent: View>: ViewModifier {
+
+    enum NavigationType {
+        case fullscreen
+        case push
+        case sheet
```

Comment:
I've added them all. But this is untested and it may result in weird views, especially for partial sheet. This is because we apply this modifier to the view in the list, not to the list as a whole.